src/main/java/nl/tudelft/jpacman/Launcher.java

```diff
@@ -171,12 +250,119 @@ private Player getSinglePlayer(final Game game) {
 	/**
 	 * Creates and starts a JPac-Man game.
 	 */
-	public void launch() {
+	public void launch(boolean askForName) {
 		game = makeGame();
-		PacManUiBuilder builder = new PacManUiBuilder().withDefaultButtons();
+		game.addObserver(this);
+		PacManUiBuilder builder;
+		if(hasMultipleLevels()){
+			builder = new PacManUiBuilder().withLevelsButtons();
+		}else{
+			builder = new PacManUiBuilder().withDefaultButtons();
+		}
 		addSinglePlayerKeys(builder, game);
 		pacManUI = builder.build(game);
 		pacManUI.start();
+		if(askForName){
+			askForPlayerName(pacManUI);
+		}
+	}
+
+	/**
+	 * Display a pop-up window to ask the player his name
+	 * 
+	 * @param UI The User Interface the game uses
+	 */
+	private void askForPlayerName(PacManUI UI) {
+		String name = null;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Use equals() to compare strings instead of '==' or '!='](https://www.codacy.com/app/mikis-millemaci/jpacman-framework/file/2535620369/issues/source?bid=3256381&fileBranchId=3284358#l277)